### ISSUE TITLE
Code running result

### ISSUE THREAD

**Issue by zwyzhr** [yingmao/Quantum-Generative-Adversarial-Network#1]
Why is the result I run is far from the paper？
![image](https://user-images.githubusercontent.com/89520974/140749654-6190a14e-4aaf-4ed0-8e01-029c64e5bb5b.png)
![$(@OG%)DIE0NH2HI J1QH40](https://user-images.githubusercontent.com/89520974/140749849-ed3244ed-bc84-435f-8e80-d4fcb3d8aa60.png)
Results in the paper:
![3J ~) TIA(0(J5`92%~Q5KR](https://user-images.githubusercontent.com/89520974/140749875-357b8e07-92e0-435b-8633-4794dc97d055.png)

**Comment by Anand270294** [comment]
Hi, I am currently also experencing the same issue with the generated images and also the discriminator loss seems to be approaching inf after 30 epochs.

**Comment by zwyzhr** [comment]
> Hi, I am currently also experencing the same issue with the generated images and also the discriminator loss seems to be approaching inf after 30 epochs.

So your result is the same as mine？

**Comment by Anand270294** [comment]
Yeah, it is, I cant seem to get the results shown the in paper but maybe it because I might be doing something wrong. So I am trying to recreate the code properly and see if it works. Apparently the numbers cant seem to be generated in my case.

**Comment by Samuelstein1224** [comment]
Hi, the experiment settings for the notebook are set to be PCA-2 Dim (i.e. 5 qubit total, see variable k) and only one set of labels. To change this, you can append the line : 

> valid_labels += train_labels == 3 

prior to data processing.
Furthermore, increase k to 9, and then PCA to 4 Dimensions, and change the train_labels = train_labels[:10000] section to include all data. 

If this does not work please post an update and I will investigate it.

**Comment by SamYJ2606** [comment]
Hi, can you please clarify what are you referring to which part of the code are you referring to with regards to the PCA dimension? There is no way to alter such a parameter other than the k value according to how the code is written.

**Comment by Samuelstein1224** [comment]
Within the code, there is a section with PCA. The n_dimensions can be changed, and accordingly with the k value. So if you use 4 PCA dimensions, k should be 9 (i.e. 9 qubits), or 5 depending on which encoding method you are using.

**Comment by SamYJ2606** [comment]
I have changed the n_dimensions to 6, and the results are still the same. May I know what was your n_dimension when you managed to obtain different results?

**Comment by BassantTolba1234** [comment]
Dear Prof,
please if you get the same result of the paper, can you kindly share it with me?
I'm waiting for your reply.
Thanks a lot.

**Comment by BassantTolba1234** [comment]
Dear Prof, @Anand270294 @SamYJ2606 @Samuelstein1224 @zwyzhr 
please if you get the same result of the paper, can you kindly share it with me?
I'm waiting for your reply.
Thanks a lot.

**Comment by BassantTolba1234** [comment]
Dear Prof, @Anand270294 @SamYJ2606 @Samuelstein1224 @zwyzhr Is the proposed code used for training only ?.. please can you kindly explain and provide the code for testing please ?

**Comment by BassantTolba1234** [comment]
Dear Prof, @Anand270294 @SamYJ2606 @Samuelstein1224 @zwyzhr Is the proposed code used for training only ?.. please can you kindly share the code version for running this code on a real quantum computer not a simulator?
